src/norm/sqlite.nim

```diff
@@ -91,10 +91,23 @@ proc createTables*[T: Model](dbConn; obj: T) =
       fkGroups.add "FOREIGN KEY($#) REFERENCES $#($#)" %
         [obj.col(fld), typeof(get val.model).table, typeof(get val.model).col("id")]
 
+    when obj.dot(fld).hasCustomPragma(fk):
+      const pragVal = $(obj.dot(fld).getCustomPragmaVal(fk))
+      const fkTargetIsModel = isModel(none(typeof(obj.dot(fld).getCustomPragmaVal(fk))))
+      when val is SomeInteger and fkTargetIsModel:
+        fkGroups.add "FOREIGN KEY ($#) REFERENCES $#(id)" % [fld, $pragVal]
+      else:
+        {.error: "Pragma fk must be used on SomeInteger field and target a Model." .}
+
     colGroups.add colShmParts.join(" ")
 
   let qry = "CREATE TABLE IF NOT EXISTS $#($#)" % [T.table, (colGroups & fkGroups).join(", ")]
 
+  # Enable foreign key
```

Comment:
I'm not sure this is necessary. Foreign keys are working just fine with SQLite right now, without this explicit pragma.

Reply:
I agree that with or without, I haven't noticed any change but the official doc of sqlite3 says it's necessary to enable it by hand : 

https://sqlite.org/pragma.html#pragma_foreign_keys 
 > As of SQLite version 3.6.19, the default setting for foreign key enforcement is OFF. However, that might change in a future release of SQLite. The default setting for foreign key enforcement can be specified at compile-time using the SQLITE_DEFAULT_FOREIGN_KEYS preprocessor macro. To minimize future problems, applications should set the foreign key enforcement flag as required by the application and not depend on the default setting. 

https://sqlite.org/foreignkeys.html (chapter 2)
 > Assuming the library is compiled with foreign key constraints enabled, it must still be enabled by the application at runtime, using the PRAGMA foreign_keys command

It could still be done by the user at the application level ?

Reply:
> It could still be done by the user at the application level ?

If you absolutely must, I think you can just add `exec sql "PRAGMA foreign_keys = ON"` into you app ¯\_(ツ)_/¯

Reply:
Okay, I'm fine with that :smiley: